node/node.go

```diff
@@ -801,7 +799,33 @@ func (n *Node) superviseManager(ctx context.Context, securityConfig *ca.Security
 		timer := time.NewTimer(16 * time.Second)
 		select {
 		case <-timer.C:
-			log.G(ctx).Warn("failed to get worker role after manager stop, restarting manager")
+			if wasRemoved {
+				// We need to be extra careful about restarting
+				// the manager. It may cause the node to
+				// wrongly join under a new Raft ID.
+				log.G(ctx).Warn("failed to get worker role after manager stop, forcing certificate renewal")
+				timer.Reset(16 * time.Second)
```

Comment:
Non-blocking nitpick:  Can we use a constant for the 16 seconds?  (also just to satisfy my curiosity, why specifically 16 seconds in this timer and the previous timer)?